integration_tests/src/main/python/join_test.py

```diff
@@ -255,6 +265,19 @@ def do_join(spark):
     conf = copy_and_update(allow_negative_scale_of_decimal_conf, {'spark.rapids.sql.batchSizeBytes': batch_size})
     assert_gpu_and_cpu_are_equal_collect(do_join, conf=conf)
 
+# local sort because of https://github.com/NVIDIA/spark-rapids/issues/84
+# After 3.1.0 is the min spark version we can drop this
+@ignore_order(local=True)
+@pytest.mark.order(1) # at the head of xdist worker queue if pytest-order is installed
+@pytest.mark.parametrize('data_gen', single_level_array_gens + map_string_string_gen + [all_basic_struct_gen], ids=idfn)
+@pytest.mark.parametrize('batch_size', ['100', '1g'], ids=idfn) # set the batch size so we can test multiple stream batches
+def test_cartesian_join_nestedtype_nonkey(data_gen, batch_size):
```

Comment:
Cartesian joins don't have a key, it's just a nested for loop to generate every row combination between the two tables.  This test should be combined with the previous test to test against all types, nested or not.

Reply:
combined with previous test